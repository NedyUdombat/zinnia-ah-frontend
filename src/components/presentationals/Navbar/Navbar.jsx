import React from 'react';
import PropTypes from 'prop-types';

import { Dropdown } from 'semantic-ui-react';
import Avatar from '../Avatar/Avatar';
import './Navbar.scss';

const Navbar = ({ url, className, children }) => {
  return (
    <div>
      <div className="navbar">
        <div>
          <h1 className="brand">Authors Haven</h1>
        </div>
        {url && (
          <div className="navbar-avatar">
            <Avatar className={className} url={url} />
            <Dropdown className="dropdown">
              <Dropdown.Menu>
                <Dropdown.Item>
                  <a>Logout</a>
                </Dropdown.Item>
              </Dropdown.Menu>
            </Dropdown>
          </div>
        )}
        {children && children}
      </div>
    </div>
  );
};

Navbar.propTypes = {
  children: PropTypes.any,
  className: PropTypes.string,
  url: PropTypes.string,
  children: PropTypes.any,
};

export default Navbar;
