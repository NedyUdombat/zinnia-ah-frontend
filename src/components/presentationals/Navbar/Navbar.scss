@import url('https://fonts.googleapis.com/css?family=Lato:400,700|Lora:400,700|Sacramento');
@import '../../../assets/variables';

.navbar {
  align-items: center;
  display: flex;
  height: 6.25rem;
  justify-content: space-between;
  padding: 0% 6%;

  .brand {
    color: $primary-black;
    font-family: 'Sacramento', cursive;
    font-size: 2.25rem;
    line-height: 3.3125rem;
  }

  .navbar-avatar {
    align-items: center;
    display: flex;
    margin-left: 1rem;
  }

  .navbar-avatar .left {
    padding: 0;
    margin: 0;
  }

  .navbar-avatar .active {
    background: transparent;
    color: inherit;
  }
}

.sm-hide {
  display: block;
}

.lg-hide {
  display: none;
}

.form-control {
  width: 100%;
  height: 3.125rem
}

@media only screen and (max-width: 56.25rem) {
  .navbar {
    justify-content: space-between;
  }

  .sm-hide {
    display: none;
  }
  .nav-items .hidden-link {
    position: fixed;
    bottom: 0;
    right: 0;
    margin: 0 1rem 4rem 0;
    z-index: 10000;
  }
  .write-btn {
    background: $primary-black;
    width: fit-content;
    padding: 1.5rem;
    border-radius: 100%;
    color: $primary-white;
  }


  .hidden-link:hover, .write-btn:hover {
    cursor: pointer;
  }

.lg-hide {
  display: block;
}
}

.ui.menu {
  box-shadow: none;
  background: transparent;
  border: none;
}

.ui.button {
  margin: 0;
}
