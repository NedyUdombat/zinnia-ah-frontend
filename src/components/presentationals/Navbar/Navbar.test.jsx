import React from 'react';
import { shallow } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';

import { Navbar } from './Navbar';

describe('Navbar', () => {
  const props = { autoLogin: jest.fn() };
  it('should render correctly', () => {
    const component = shallow(
      <Navbar {...props} profileUrl="http://dummyimage.com" className="" />,
    );
    expect(component).toMatchSnapshot();
  });
});
