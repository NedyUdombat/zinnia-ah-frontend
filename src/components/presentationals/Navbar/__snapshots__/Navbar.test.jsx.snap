// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render correctly 1`] = `
<div>
  <div
    className="navbar"
  >
    <div>
      <h1
        className="brand"
      >
        <Link
          style={
            Object {
              "color": "inherit",
            }
          }
          to="/"
        >
          Authors Haven
        </Link>
      </h1>
    </div>
    <div
      className="sm-hide"
    >
      <div
        className="nav-items"
      >
        <form
          className="search-bar"
          onSubmit={[Function]}
        >
          <Input
            className="form-control"
            data-test="search-input"
            icon="search"
            iconPosition="left"
            name="keyword"
            onChange={[Function]}
            placeholder="Search Authors' Haven"
            type="search"
            value=""
          />
        </form>
        <Link
          to="/editor"
        >
          <Button
            className="btn btn-dark w-10"
            type="submit"
            value="WRITE"
          />
        </Link>
        <Link
          style={
            Object {
              "margin": "0 15px",
            }
          }
          to={
            Object {
              "pathname": "/login",
              "state": Object {
                "from": undefined,
              },
            }
          }
        >
          <Button
            className="btn-dark"
            value="LOGIN"
          />
        </Link>
      </div>
    </div>
  </div>
</div>
`;
